Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m

```diff
@@ -25,81 +26,184 @@
 % permissions and limitations under the License.                        %
 %-----------------------------------------------------------------------%
 
-% Three variables: 1) passive stiffness 
-%                  2) active max torque 
-%                  3) hop timing
-% See buildHopperFromSolution() for more details.
-numVariables = 3; 
+% CHOOSE PROBLEM: 'one_hop' or 'three_hops'
+problem = 'one_hop';
+switch problem
+    
+    % Optimize the height of one hop. The single optimization variable 
+    % determines when to simultaneously release a loaded passive device and 
+    % activate the hopper vastus muscle (see CONSTRUCTCONTROLS below). This
+    % simple univariate problem uses FMINBND from the MATLAB Optimization 
+    % Toolbox. 
+    case 'one_hop'
+        lb = 0;
+        ub = 1;         
+        options = optimset('TolFun', 1e-2, ...
+                           'MaxIter', 10000, ...
+                           'Display', 'iter'); 
+        [x, f] = fminbnd(@(x) objective(x,problem),lb,ub,options);
+                     
+    % Optimize the height of a sequence of three hops. The optimization
+    % variables decide to switch the actuators on and off to load and 
+    % release the passive device with the activated muscle (see 
+    % CONSTRUCTCONTROLS below). This problem is subject to falling into
+    % undesired local minima (e.g. solutions with less than three hops) and
+    % requires an optimizer from the MATLAB Global Optimization Toolbox.
+    % Here we use the genetic algorithm optimizer, GA.
+    case 'three_hops'
+        numVars = 3; 
+        lb = 0.25*ones(numVars, 1);
+        ub = 0.75*ones(numVars, 1);
```

Comment:
Does this mean the hops happen within the first second? Consider putting units in comments for `lb` and `ub`.

Reply:
Perhaps a clear description of the design variables (i.e., the vector `x`) above could help here, since these map directly to them.

Reply:
The units are in seconds. It means the first hop happens with the first second, and then the others happen based on additions of the three variables as described in `constructControls`.